partialWaveFit/fitResult.cc

```diff
@@ -387,34 +401,49 @@ fitResult::fill(const fitResult& result,
                 const complexMatrix*            acceptedNormIntegral,    // normalization integral matrix with acceptance
                 const vector<double>*           phaseSpaceIntegral)      // normalization integral over full phase space without acceptance
 {
-	std::vector<std::complex<double>> prodAmpsEmpty;
+	if (fitParCovMatrix == nullptr)
+		fitParCovMatrix = (not result.covMatrixValid()) ? nullptr : &result.fitParCovMatrix();
+	if (fitParCovMatrixIndices == nullptr)
+		fitParCovMatrixIndices = &result.fitParCovIndices();
+	if (normIntegral == nullptr)
+		normIntegral = (result.normIntegralMatrix().nRows() == 0 and result.normIntegralMatrix().nCols() == 0) ? nullptr : &result.normIntegralMatrix();
+	if (acceptedNormIntegral == nullptr)
+		acceptedNormIntegral = (result.acceptedNormIntegralMatrix().nRows() == 0 and result.acceptedNormIntegralMatrix().nCols() == 0) ?
+		                       nullptr : &result.acceptedNormIntegralMatrix();
+	if (phaseSpaceIntegral == nullptr)
+		phaseSpaceIntegral = (result.phaseSpaceIntegralVector().size() == 0) ? nullptr : &result.phaseSpaceIntegralVector();
 	fill(result.nmbEvents(), result.normNmbEvents(), result.multibinBoundaries(), result.logLikelihood(),
-	     result.rank(), prodAmpsEmpty, result.prodAmpNames(),
-	     (fitParCovMatrix!=nullptr)? fitParCovMatrix : &result.fitParCovMatrix(),
-	     (fitParCovMatrixIndices!=nullptr)? fitParCovMatrixIndices : &result.fitParCovIndices(),
-	     (normIntegral!=nullptr)? normIntegral: &result.normIntegralMatrix(),
-	     (acceptedNormIntegral!=nullptr)? acceptedNormIntegral : &result.acceptedNormIntegralMatrix(),
-	     (phaseSpaceIntegral!=nullptr)? phaseSpaceIntegral: &result.phaseSpaceIntegralVector(),
+	     result.rank(), result.prodAmps(), result.prodAmpNames(),
+	     fitParCovMatrix,
+	     *fitParCovMatrixIndices,
+	     normIntegral,
+	     acceptedNormIntegral,
+	     phaseSpaceIntegral,
 	     result.converged(), result.hasHessian());
-	// set prodAmps directly to avoid converting vector<TComplex> -> vector<comple<double>> -> vector<TComplex>
-	_prodAmps               = result.prodAmps();
 }
 
 
 void
-fitResult::fill(const fitResult& result, const bool fillCovMatrix, const bool fillIntegralMatrices)
+fitResult::fill(const fitResult& result, bool fillCovMatrix, bool fillIntegralMatrices)
 {
-	std::vector<std::complex<double>> prodAmpsEmpty;
+	if (fillCovMatrix)
+		fillCovMatrix = result.covMatrixValid();
+	bool fillNormIntMatrix = false;
+	bool fillAccNormIntMatrix = false;
+	bool fillPhaseSpace = false;
+	if (fillIntegralMatrices){
+		fillNormIntMatrix    = result.normIntegralMatrix().nRows() != 0         or result.normIntegralMatrix().nCols() != 0;
+		fillAccNormIntMatrix = result.acceptedNormIntegralMatrix().nRows() != 0 or result.acceptedNormIntegralMatrix().nCols() != 0;
```

Comment:
`or` -> `and`?

Reply:
I think it is correct his way, because you want to interpret the matrix only as empty if rows==0 and cols ==0.

Reply:
No, a matrix with rows==0 and cols!=0 also does not contain any entries.

Reply:
Yes.
However, there is in principle a difference between a result as an empty matrix, which would also be rows==0 and cols!=0, and a result as now matrix.
My proposal as a representation for a result without matrix is rows==0 and cols==0.
Additionally, if the old result has a matrix with rows==0 and cols!=0 there is clearly something wrong.
The question here is whether the fill function should rise a warning in such a situation, which it does at the moment.

Reply:
ahh, yes, you are right.